internal/installercache/installercache_test.go

```diff
@@ -161,42 +331,42 @@ var _ = Describe("installer cache", func() {
 		Expect(os.IsNotExist(err)).To(BeTrue())
 		_, err = os.Stat(r3)
 		Expect(os.IsNotExist(err)).To(BeFalse())
+		// Now measure disk usage, we should be under the cache size
+		Expect(getUsedBytesForDirectory(manager.config.CacheDir)).To(BeNumerically("<=", manager.config.MaxCapacity))
 	})
 
-	It("when cache limit is not set eviction is skipped", func() {
-		manager.storageCapacity = 0
+	It("evicts the oldest file", func() {
 		clusterId := strfmt.UUID(uuid.New().String())
-		r1, _ := testGet("4.8", "4.8.0", clusterId, false)
-		r2, _ := testGet("4.9", "4.9.0", clusterId, false)
-		r3, _ := testGet("4.10", "4.10.0", clusterId, false)
+		r1, l1 := testGet("4.8", "4.8.0", clusterId, false)
+		r2, l2 := testGet("4.9", "4.9.0", clusterId, false)
+		r3, l3 := testGet("4.10", "4.10.0", clusterId, false)
 
-		By("verify that the no file was deleted")
+		By("verify that the oldest file was deleted")
 		_, err := os.Stat(r1)
-		Expect(os.IsNotExist(err)).To(BeFalse())
+		Expect(os.IsNotExist(err)).To(BeTrue())
 		_, err = os.Stat(r2)
 		Expect(os.IsNotExist(err)).To(BeFalse())
 		_, err = os.Stat(r3)
 		Expect(os.IsNotExist(err)).To(BeFalse())
 
+		By("verify that the links were purged")
+		manager.evict()
+		_, err = os.Stat(l1)
+		Expect(os.IsNotExist(err)).To(BeTrue())
+		_, err = os.Stat(l2)
+		Expect(os.IsNotExist(err)).To(BeTrue())
+		_, err = os.Stat(l3)
+		Expect(os.IsNotExist(err)).To(BeTrue())
+		// Now measure disk usage, we should be under the cache size
+		Expect(getUsedBytesForDirectory(manager.config.CacheDir)).To(BeNumerically("<=", manager.config.MaxCapacity))
 	})
 
 	It("extracts from the mirror", func() {
```

Comment:
Where is this test actually testing anything related to the mirror? This one and the following test look almost exactly the same (at least the verification bits do).
Can we make sure this is testing what it is supposed to be testing?

Reply:
You know what, you're right...
Again these methods were already in the tests and I did not look too deeply at them.

But we can basically delete one of them, all we are proving is that the release mirror parameter gets passed to the openshift extraction code. No value in testing both paths.